src/refract/RenderJSONVisitor.cc

```diff
@@ -63,11 +64,42 @@ namespace refract
         e.content(*this);
     }
 
+    template<typename T>
+    const bool isNullable(const T& e) {
+        IElement::MemberElementCollection::const_iterator ta = e.attributes.find("typeAttributes");
+        
+        if (ta == e.attributes.end()) {
+            return false;
+        }
+
+        ArrayElement* attrs = TypeQueryVisitor::as<ArrayElement>((*ta)->value.second);
+
+        if (!attrs) {
+            return false;
+        }
+
+        for (ArrayElement::ValueType::const_iterator it = attrs->value.begin() ; it != attrs->value.end() ; ++it ) {
+            StringElement* attr = TypeQueryVisitor::as<StringElement>(*it);
+            if (!attr) {
+                continue;
+            }
+            if (attr->value == "nullable") {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
     void RenderJSONVisitor::visit(const MemberElement& e) {
         RenderJSONVisitor renderer;
 
         if (e.value.second) {
-            renderer.visit(*e.value.second);
+            if (isNullable(e) && e.value.second->empty() == true) {
```

Comment:
I am not asking you to remove the whole check. I am asking you to change the line to the following:

```cpp
if (isNullable(e) && e.value.second->empty()) {
```